### PR TITLE
feat: dashboard geojson map

### PR DESCRIPTION
**Pull request by ktx-vaidehi**
- #4825 

## Release Notes

- **New Features**
  - Introduced new `MapsRenderer` component for enhanced map visualizations.
  - Added support for selecting map types in the configuration panel.
  - New dashboard components for managing map queries and data.
  - Added new fields for map configurations, including names and values.

- **Improvements**
  - Enhanced chart selection options, including new map-related types.
  - Updated localization for better support of map functionalities.
  - Improved handling of field aliases in the panel schema.
  - Enhanced functionality for adding map names and values within the dashboard.
  - Refined drag-and-drop functionality for map fields.
  - Enhanced data validation and error handling in map-related components.

- **Bug Fixes**
  - Improved error handling and data validation in map-related components.

- **Chores**
  - Updated component names and structure for consistency.
  - Enhanced internal methods for better data handling and processing.

### CODE REVIEW

**Comment by gaby** [comment]
@ktx-vaidehi @ktx-kirtan This is such a good feature, assuming it uses echarts map without having to use a map service?

**Review by ktx-kirtan — APPROVED** [review]

**Comment by ktx-kirtan** [comment]
@coderabbitai review

File: web/src/components/dashboards/addPanel/DashboardMapsQueryBuilder.vue [hunk 1]
```diff
@@ -0,0 +1,869 @@
...
+                    filled
+                    data-test="dashboard-name-item-input"
+                    :label="t('common.label')"
+                    v-model="
+                      dashboardPanelData.data.queries[
+                        dashboardPanelData.layout.currentQueryIndex
+                      ].fields.name.label
+                    "
+                    :rules="[(val: any) => val > 0 || 'Required']"
+                  />
```
[the remark this answers is not in the document]
Comment (gaby) [reply web/src/components/dashboards/addPanel/DashboardMapsQueryBuilder.vue:108]: @ktx-vaidehi I think this is a needed change?

[the remark this answers is not in the document]
Comment (ktx-vaidehi) [reply web/src/components/dashboards/addPanel/DashboardMapsQueryBuilder.vue:108]: > @ktx-vaidehi I think this is a needed change?

Yes.

File: web/src/components/dashboards/PanelSchemaRenderer.vue [hunk 2]
```diff
@@ -768,7 +797,7 @@
             await getFoldersList(store);
           }
           const folderId = store.state.organizationData.folders.find(
-            (folder: any) => folder.name == drilldownData.data.folder
+            (folder: any) => folder.name == drilldownData.data.folder,
```
[the remark this answers is not in the document]
Comment (ktx-vaidehi) [reply web/src/components/dashboards/PanelSchemaRenderer.vue:800]: done changes

File: web/src/components/dashboards/addPanel/ConfigPanel.vue [hunk 3]
```diff
@@ -240,6 +242,22 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
+        :options="mapTypeOptions"
+        dense
+        :label="t('dashboard.mapTypeLabel')"
+        class="showLabelOnTop"
+        stack-label
+        emit-value
+        data-test="dashboard-config-map-type"
+      >
+      </q-select>
+
```
[the remark this answers is not in the document]
Comment (ktx-vaidehi) [reply web/src/components/dashboards/addPanel/ConfigPanel.vue:258]: added

File: web/src/utils/dashboard/convertMapsData.ts [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+  { name: "Israel", alpha2: "IL", alpha3: "ISR" },
+  { name: "Italy", alpha2: "IT", alpha3: "ITA" },
+  { name: "Jamaica", alpha2: "JM", alpha3: "JAM" },
+  { name: "Japan", alpha2: "JP", alpha3: "JPN" },
+  { name: "Jordan", alpha2: "JO", alpha3: "JOR" },
+  { name: "Kazakhstan", alpha2: "KZ", alpha3: "KAZ" },
+  { name: "Kenya", alpha2: "KE", alpha3: "KEN" },
+  { name: "Kiribati", alpha2: "KI", alpha3: "KIR" },
+  { name: "Korea (North)", alpha2: "KP", alpha3: "PRK" },
+  { name: "Korea (South)", alpha2: "KR", alpha3: "KOR" },
```
[the remark this answers is not in the document]
Comment (ktx-vaidehi) [reply web/src/utils/dashboard/convertMapsData.ts:107]: the data has different name sometimes

File: web/src/components/dashboards/PanelSchemaRenderer.vue [hunk 5]
```diff
@@ -179,6 +192,10 @@ const GeoMapRenderer = defineAsyncComponent(() => {
   return import("@/components/dashboards/panels/GeoMapRenderer.vue");
 });
 
+const MapsRenderer = defineAsyncComponent(() => {
+  return import("@/components/dashboards/panels/MapsRenderer.vue");
+});
```
[the remark this answers is not in the document]
Comment (ktx-vaidehi) [reply web/src/components/dashboards/PanelSchemaRenderer.vue:197]: this is not needed for now

**Pull request merged**